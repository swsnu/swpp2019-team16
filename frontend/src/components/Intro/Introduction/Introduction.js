import clns from 'classnames';
import FullPage from '@fullpage/react-fullpage';
import PropTypes from 'prop-types';
import React from 'react';
import { makeStyles } from '@material-ui/styles';
import theme from '../../../lib/styles/theme';
import Typography from '@material-ui/core/Typography';
import Button from '../../common/Button';

const useWhatIsSectionStyles = makeStyles({
  root: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'flex-start',
    alignItems: 'center',
    paddingTop: '0',
    paddingBottom: '0',

    backgroundColor: theme.palette.primary.dark,
    color: theme.palette.primary.contrastText,

    paddingLeft: '10px',
    paddingRight: '10px',
  },
  title: {
    marginLeft: '200px',
    marginRight: '200px',
    marginBottom: '24px',
  },
  content: {
    marginLeft: '200px',
    marginRight: '200px',
    fontSize: '24px',
    lineHeight: '28px',
  },
  lastContent: {
    marginLeft: '200px',
    marginRight: '200px',
    fontSize: '24px',
    lineHeight: '28px',
    marginBottom: '50px',
  },
  getStartedButton: {
    marginLeft: '220px',
    marginRight: '200px',
    marginBottom: '100px',
  },
});

WhatIsSection.propTypes = {
  onStart: PropTypes.func.isRequired,
};

function WhatIsSection({ onStart }) {
  const styles = useWhatIsSectionStyles();
  return (
    <div className={clns('section', styles.root)}>
      <div className={styles.title}>
        <Typography variant="h1">Ya-Ta comes to rescue!</Typography>
      </div>
      <div className={styles.content}>
        <Typography variant="h2">
          Ya-Ta helps students who need to urgently get to class
        </Typography>
      </div>
      <div className={styles.content}>
        <Typography variant="h2">
          through luring taxi drivers with the incentives
        </Typography>
      </div>
      <div className={styles.lastContent}>
        <Typography variant="h2">
          and save money through a carpool system
        </Typography>
      </div>
      <div className={styles.getStartedButton}>
        <Button size={'large'} color={'secondary'} onClick={onStart}>
          Get Started
        </Button>
      </div>
    </div>
  );
}

const useSubSectionStyles = makeStyles({
  root: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'flex-start',
    alignItems: 'center',
    paddingTop: '0',
    paddingBottom: '0',

    backgroundColor: theme.palette.white,
    color: theme.palette.text.primary,

    paddingLeft: '10px',
    paddingRight: '10px',
  },
  title: {
    marginLeft: '200px',
    marginRight: '200px',
    marginBottom: '24px',
  },
  subtitle: {
    marginLeft: '200px',
    marginRight: '200px',
    marginBottom: '50px',
  },
  getStartedButton: {
    marginLeft: '220px',
    marginRight: '200px',
    marginBottom: '100px',
  },
});

SubSection.propTypes = {
  onStart: PropTypes.func.isRequired,
};

function SubSection({ onStart }) {
  const styles = useSubSectionStyles();
  return (
    <div className={clns('section', styles.root)}>
      <div className={styles.title}>
        <Typography variant="h1">The clock is ticking.</Typography>
      </div>
      <div className={styles.subtitle}>
        <Typography variant="h2">
          The professor will check the attendance as soon as class starts
        </Typography>
      </div>
      <div className={styles.getStartedButton}>
        <Button size={'large'} onClick={onStart}>
          Get Started
        </Button>
      </div>
    </div>
  );
}

const useMainSectionStyles = makeStyles({
  root: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'flex-start',
    alignItems: 'center',
    paddingTop: '0',
    paddingBottom: '0',

    backgroundColor: theme.palette.primary.main,
    color: theme.palette.white,

    paddingLeft: '10px',
    paddingRight: '10px',
  },
  title: {
    marginLeft: '200px',
    marginRight: '200px',
    marginBottom: '24px',
  },
  subtitle: {
    marginLeft: '200px',
    marginRight: '200px',
    marginBottom: '50px',
  },
  getStartedButton: {
    marginLeft: '220px',
    marginRight: '200px',
    marginBottom: '100px',
  },
});

MainSection.propTypes = {
  onStart: PropTypes.func.isRequired,
};

function MainSection({ onStart }) {
  const styles = useMainSectionStyles();
  return (
    <div className={clns('section', styles.root)}>
      <div className={styles.title}>
        <Typography variant="h1">
          No more waiting long line of SNU shuttle bus!
        </Typography>
      </div>
      <div className={styles.subtitle}>
        <Typography variant="h2">
          On a bad day, it may take around 20~30 minutes to get to class
        </Typography>
      </div>
      <div className={styles.getStartedButton}>
        <Button size={'large'} color={'secondary'} onClick={onStart}>
          Get Started
        </Button>
      </div>
    </div>
  );
}

Introduction.propTypes = {
  onStart: PropTypes.func.isRequired,
};

function Introduction({ onStart }) {
  return (
    <FullPage
      scrollingSpeed={1000}
      render={({ state, fullpageApi }) => {
        return (
          <FullPage.Wrapper>
            <MainSection onStart={onStart} />
            <SubSection onStart={onStart} />
            <WhatIsSection onStart={onStart} />
          </FullPage.Wrapper>
        );
      }}
    />
  );
}

export default Introduction;
