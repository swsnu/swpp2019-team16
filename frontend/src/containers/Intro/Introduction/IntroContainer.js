import React, { useCallback } from 'react';
import { withRouter } from 'react-router-dom';
import Introduction from '../../../components/Intro/Introduction';

function IntroContainer({ history }) {
  const onStart = useCallback(() => {
    history.push('/request');
  }, [history]);
  return <Introduction onStart={onStart} />;
}
export default withRouter(IntroContainer);
