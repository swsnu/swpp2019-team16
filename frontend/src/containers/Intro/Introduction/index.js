import IntroContainer from './IntroContainer';

export default IntroContainer;
