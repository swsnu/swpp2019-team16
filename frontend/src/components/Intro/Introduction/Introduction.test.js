import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Introduction from './Introduction';

jest.mock('@fullpage/react-fullpage', () => ({ children }) => (
  <div>{children}</div>
));

describe('<Introduction />', () => {
  it('SHOULD match with snapshot', async () => {
    const { container } = render(<Introduction onStart={() => {}} />);
    expect(container).toMatchSnapshot();
  });
});
