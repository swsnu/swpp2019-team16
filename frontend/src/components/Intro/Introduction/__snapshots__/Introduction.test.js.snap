// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Introduction /> SHOULD match with snapshot 1`] = `
<div>
  <div />
</div>
`;
